### PR TITLE
Fix issues that prevented runtime to be properly added to the component.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: core/src/main/java/io/ap4k/Resources.java [hunk 1]
```diff
@@ -90,6 +88,15 @@ public void add(String group, HasMetadata metadata) {
     groups.get(group).addToItems(metadata);
   }
 
+  /**
+   * Add a {@link Visitor} to the specified resource explicit group.
```
Comment (cmoulliard) [inline core/src/main/java/io/ap4k/Resources.java]: Should be renamed to `@Link Decorator` ?

File: core/src/main/java/io/ap4k/Resources.java [hunk 2]
```diff
@@ -90,6 +88,15 @@ public void add(String group, HasMetadata metadata) {
...
+   * Explicit groups are groups that only get resources and decorators that have been explicitly specified.
```
Comment (cmoulliard) [inline core/src/main/java/io/ap4k/Resources.java]: What will be then happen if they are `explicitly` declared ? This code source doc should be improved

File: core/src/main/java/io/ap4k/config/ConfigKey.java [hunk 3]
```diff
@@ -44,4 +46,19 @@ public String getName() {
   public T getDefaultValue() {
     return defaultValue;
   }
+
```
Comment (cmoulliard) [inline core/src/main/java/io/ap4k/config/ConfigKey.java]: Why do we need `equals(Object o) and hasCode()` ?

File: frameworks/spring-boot/src/main/java/io/ap4k/spring/configurator/SetSpringBootRuntime.java [hunk 4]
```diff
@@ -18,17 +18,16 @@
 
 import io.ap4k.config.ConfigKey;
 import io.ap4k.config.Configurator;
-import io.ap4k.config.KubernetesConfigBuilder;
-import io.ap4k.deps.kubernetes.api.builder.TypedVisitor;
+import io.ap4k.config.KubernetesConfigFluent;
 
-public class SetSpringBootRuntime extends Configurator<KubernetesConfigBuilder> {
+public class SetSpringBootRuntime extends Configurator<KubernetesConfigFluent> {
```
Comment (cmoulliard) [inline frameworks/spring-boot/src/main/java/io/ap4k/spring/configurator/SetSpringBootRuntime.java]: Have you documented within the contrib guide the difference between `KubernetesConfigBuilder ` and `KubernetesConfigFluent ` and when they should or not been used ?

File: core/src/main/java/io/ap4k/config/ConfigKey.java (hunk 3 above)
Comment (iocanel) [reply core/src/main/java/io/ap4k/config/ConfigKey.java]: In order to exclude `defaultValue, which can't participate in the key.

File: doc/src/main/resources/sequence.wsd [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+opt generation phase
+
+    c->+p: notify processing over
+    p->+s: close
+    s-->+s: apply configurators
+    s-->+s: generate config
+    s->+h: send config
+    h->+s: generate model
+    s-->+s: apply docorators
```
Comment (cmoulliard) [inline doc/src/main/resources/sequence.wsd]: Should be `decorators`

File: design.md [hunk 6]
```diff
@@ -1,16 +1,79 @@
...
+
+The section below describes the core parts of AP4K and tries to describe how they work together.
+
+### Visitor
+Refers to the the `Gang of Four` visitor pattern. As Kubernetes resources are deeply nested we are extensively using the 
+`visitor` pattern  to perform modifications to those resources without having to programmatically traverse this complex
+structure.
+
+### Model
+With kubernetes/openshift `model` or just `model` we refer to the java object representation of the Kubernetes/Openshift
```
Comment (cmoulliard) [inline design.md]: I propose to rephrase this sentence as `A `model` refers to java objects which describes the Kubernetes, Custom Resource Definition or OpenShift resources/objects.`

File: design.md [hunk 7]
```diff
@@ -50,6 +138,20 @@ It's different than a `configurator` in the sense that it operator on the actual
 | AddLabel                      | ObjectMeta     | Add a label to the all metadata.                       |
 | AddAzureFileVolume            | PodSpec        | Add an Azure File volume to the Pod spec.              |
 
+#### Why do we need both Configurators and Decorators?
+
+The kubernetes `model` is very complex and deeply nested object structure and for a good reason: `It needs to fit to every signle deployment use case out there.`
```
Comment (cmoulliard) [inline design.md]: `signle` should be `single`

**Pull request merged**